datagateway_api/src/resources/entities/entity_map.py

```diff
@@ -106,17 +68,25 @@ def create_entity_models():
                 or relationship_class.direction.name == "ONETOONE"
             ):
                 params[relationship_name] = {
-                    "$ref": f"#/components/schemas/{relationship_name.strip('_')}",
+                    "$ref": "#/components/schemas/"
+                    f"{relationship_name.strip('_').upper()}",
                 }
             if (
                 relationship_class.direction.name == "MANYTOMANY"
                 or relationship_class.direction.name == "ONETOMANY"
             ):
+                entity_underscore_strip = relationship_name.strip("_")
+                if entity_underscore_strip[-1] == "s":
```

Comment:
Again, probably useful to specify in a comment that this is checking for plurals

Reply:
As above